samples/bluetooth/iso_connected_benchmark/src/main.c

```diff
@@ -0,0 +1,1041 @@
+};
+
+static bool data_cb(struct bt_data *data, void *user_data)
+{
+	char *name = user_data;
+	uint8_t len;
+
+	switch (data->type) {
+	case BT_DATA_NAME_SHORTENED:
+	case BT_DATA_NAME_COMPLETE:
+		len = MIN(data->data_len, DEVICE_NAME_LEN - 1);
+		memcpy(name, data->data, len);
+		name[len] = '\0';
+		return false;
+	default:
+		return true;
+	}
+}
+
+static int start_scan(void)
+{
+	int err;
+
+	err = bt_le_scan_start(BT_LE_SCAN_ACTIVE, NULL);
+	if (err != 0) {
+		LOG_ERR("Scan start failed: %d", err);
+		return err;
+	}
+
+	LOG_INF("Scan started");
+
+	return 0;
+}
+
+static int stop_scan(void)
+{
+	int err;
+
+	err = bt_le_scan_stop();
+	if (err != 0) {
+		LOG_ERR("Scan stop failed: %d", err);
+		return err;
+	}
+	LOG_INF("Scan stopped");
+
+	return 0;
+}
+
+static void scan_recv(const struct bt_le_scan_recv_info *info,
+		      struct net_buf_simple *buf)
+{
+	char le_addr[BT_ADDR_LE_STR_LEN];
+	char name[DEVICE_NAME_LEN];
+
+	if (advertiser_found) {
+		return;
+	}
+
+	(void)memset(name, 0, sizeof(name));
+
+	bt_data_parse(buf, data_cb, name);
+
+	if (strncmp(DEVICE_NAME, name, strlen(DEVICE_NAME))) {
+		return;
+	}
+
+	bt_addr_le_to_str(info->addr, le_addr, sizeof(le_addr));
+
+	LOG_INF("Found peripheral with address %s (RSSI %i)",
+		le_addr, info->rssi);
+
+
+	bt_addr_le_copy(&adv_addr, info->addr);
+	advertiser_found = true;
+	k_sem_give(&sem_adv);
+}
+
+static struct bt_le_scan_cb scan_callbacks = {
+	.recv = scan_recv,
+};
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	if (err != 0 && role == ROLE_CENTRAL) {
+		LOG_INF("Failed to connect to %s: %d", addr, err);
+
+		bt_conn_unref(default_conn);
+		default_conn = NULL;
+		return;
+	} else if (role == ROLE_PERIPHERAL) {
+		default_conn = conn;
+	}
+
+	LOG_INF("Connected: %s", addr);
+
+	k_sem_give(&sem_connected);
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	LOG_INF("Disconnected: %s (reason 0x%02x)", addr, reason);
+
+	default_conn = NULL;
+
+	k_sem_give(&sem_disconnected);
+}
+
+static struct bt_conn_cb conn_callbacks = {
+	.connected = connected,
+	.disconnected = disconnected,
+};
+
+static size_t get_chars(char *buffer, size_t max_size)
+{
+	size_t pos = 0;
+
+	while (pos < max_size) {
+		char c = console_getchar();
+
+		if (c == '\n' || c == '\r') {
+			break;
+		}
+		printk("%c", c);
+		buffer[pos++] = c;
+	}
+	printk("\n");
+	buffer[pos] = '\0';
+
+	return pos;
+}
+
+static int parse_rtn_arg(struct bt_iso_chan_io_qos *qos)
+{
+	char buffer[4];
+	size_t char_count;
+	uint64_t rtn;
+
+	printk("Set RTN (current %u, default %u)\n",
+	       qos->rtn, DEFAULT_CIS_RTN);
+
+	char_count = get_chars(buffer, sizeof(buffer) - 1);
+
+	if (char_count == 0) {
+		return DEFAULT_CIS_RTN;
+	}
+
+	rtn = strtoul(buffer, NULL, 0);
+	if (rtn > 16) {
```

Comment:
This limit as well as the interval max/min, latency max, SDU interval, are they defined somewhere?